gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/TailLocalStep.java

```diff
@@ -54,7 +54,7 @@ protected S map(final Traverser.Admin<S> traverser) {
                         start instanceof Collection ? ((Collection) start).size() :
                                 start instanceof Path ? ((Path) start).size() :
                                         start instanceof Iterable ? IteratorUtils.count((Iterable) start) :
-                                                this.limit;
+                                                IteratorUtils.count(IteratorUtils.asIterator(start));
```

Comment:
is this meant as some sort of safety? `RangeLocalStep.applyRange()` seems to satisfy the issue, no? no exception thrown here:

```
gremlin> g.inject([1,2,3],[1],10).tail(local)
==>3
==>1
==>10
```

Reply:
also to handle arrays as input
```
gremlin> g.inject([1,2,3] as int[], [1]).tail(local,1)
==>[1, 2, 3]
==>1
```